sdks/python/apache_beam/transforms/external_test.py

```diff
@@ -246,6 +248,59 @@ def test_java_expansion(self):
     finally:
       server.kill()
 
+  def test_java_expansion_dataflow(self):
+    if not self.expansion_service_jar:
+      raise unittest.SkipTest('No expansion service jar provided.')
+
+    from apache_beam.runners.dataflow.internal.apiclient import DataflowApplicationClient
+
+    with patch.object(
+        DataflowApplicationClient, 'create_job') as mock_create_job:
+
+      # The actual definitions of these transforms is in
+      # org.apache.beam.runners.core.construction.TestExpansionService.
+      TEST_FILTER_URN = "pytest:beam:transforms:filter_less_than"
+
+      # Run as cheaply as possible on the portable runner.
+      options = PipelineOptions(
+          runner='DataflowRunner',
+          project='dummyproject',
+          temp_location='gs://dummybucket/',
+          experiments=['beam_fn_api', 'use_unified_worker',
+                       'cross_language_pipeline'])
+
+      try:
+        # Start the java server and wait for it to be ready.
```

Comment:
Seems it'd be worth factoring this out into a context (including the expansion service jar test above).